packages/provider/src/RelaySelectionManager.ts

```diff
@@ -76,8 +76,8 @@ export class RelaySelectionManager {
             relayInfo
           }
         } else {
-          // TODO: do not throw! The preferred relay may be removed since.
-          throw new Error('Could not find register event for the winning preferred relay')
+          this.logger.error('Could not find registration info in the RelayRegistrar for the selected preferred relay')
```

Comment:
Well, I am not re-writing the old code that was based on wrong assumptions now. I think this is a valid question but is only relevant for a re-write. As far as I can tell this won't make an RPC call, but will just find an entry in the known relays list.

Reply:
This code is only executed for "preferred" relays because the object does not have the details and need to find them. Normal relay will have all details and not get to this branch in the first place.

Reply:
It's unclear in the code why this part handles preferred relays only